test/AWSCredentials.jl

```diff
@@ -1244,20 +1244,40 @@ end
         write(config_file, config_str)
         ini = read(Inifile(), IOBuffer(config_str))
 
+        @testset "environmental variable (DEFAULT)" begin
+            withenv("AWS_REGION" => nothing, "AWS_DEFAULT_REGION" => "us-gov-east-1") do
+                @test aws_get_region(; config=ini, profile="default") == "us-gov-east-1"
+                @test aws_get_region() == "us-gov-east-1"
+            end
+        end
+
         @testset "environmental variable" begin
             withenv("AWS_DEFAULT_REGION" => "us-gov-east-1") do
                 @test aws_get_region(; config=ini, profile="default") == "us-gov-east-1"
                 @test aws_get_region() == "us-gov-east-1"
             end
         end
 
+        @testset "environmental variable (AWS_REGION)" begin
+             withenv("AWS_REGION" => "eu-west-1", "AWS_DEFAULT_REGION" => nothing) do
+                 @test aws_get_region(; config=ini, profile="default") == "eu-west-1"
+                 @test aws_get_region() == "eu-west-1"
+             end
+
+             withenv("AWS_REGION" => "eu-west-1", "AWS_DEFAULT_REGION" => "us-gov-east-1") do
+                 @test aws_get_region(; config=ini, profile="default") == "eu-west-1"
+                 @test aws_get_region() == "eu-west-1"
+             end
+         end
```

Comment:
These might as well be all in the same `@testset`. The unmodified tests should have `"AWS_REGION" => nothing` as otherwise it could be affected by the user's set environmental variables:
```suggestion
        @testset "environmental variable (AWS_REGION)" begin
             withenv("AWS_REGION" => "eu-west-1", "AWS_DEFAULT_REGION" => nothing) do
                 @test aws_get_region(; config=ini, profile="default") == "eu-west-1"
                 @test aws_get_region() == "eu-west-1"
             end
             
             withenv("AWS_REGION" => nothing, "AWS_DEFAULT_REGION" => "us-gov-east-1") do
                @test aws_get_region(; config=ini, profile="default") == "us-gov-east-1"
                @test aws_get_region() == "us-gov-east-1"
            end

             withenv("AWS_REGION" => "eu-west-1", "AWS_DEFAULT_REGION" => "us-gov-east-1") do
                 @test aws_get_region(; config=ini, profile="default") == "eu-west-1"
                 @test aws_get_region() == "eu-west-1"
             end
         end
```